robottelo/host_helpers/satellite_mixins.py

```diff
@@ -324,3 +325,8 @@ def api_factory(self):
         if not getattr(self, '_api_factory', None):
             self._api_factory = APIFactory(self)
         return self._api_factory
+
+    def ui_factory(self, session):
```

Comment:
I think rather than passing the session we can keep the `self.ui`as attribute in the hosts.py under the satellite class